### PR TITLE
Fix visual studio compiler warnings

### PR DESCRIPTION
**Pull request by LamarMoore**
Patch boost visualc header in an attempt to eliminate compiler warnings on MSVC 2017

### CODE REVIEW

**Comment by martyngigg** [comment]
Thanks for this Lamar. Are the boost libraries still compiled with 2015 here?

**Comment by LamarMoore** [comment]
@martyngigg yes these are compiled using 2015

**Comment by martyngigg** [comment]
Cool. Could you squash the commits so that we've only got 1 version of the binaries in the history?

File: build-scripts/extras/boost/visualc.hpp.patch [hunk 1]
```diff
@@ -1,11 +1,101 @@
...
+@@ -293,11 +326,11 @@
+ #endif
  
  //
- // last known and checked version is 19.10.24629 (VC++ 2017 RC):
+-// last known and checked version is 19.10.24629 (VC++ 2017 RC):
 -#if (_MSC_VER > 1910)
-+#if (_MSC_VER > 1911)
++// last known and checked version is 19.11.25547 (VC++ 2017.4):
++#if (_MSC_VER > 1913)
```
Comment (martyngigg) [inline build-scripts/extras/boost/visualc.hpp.patch]: Are going to have to change this every time they release a new update for 2017?

**Comment by quantumsteve** [comment]
Why are we patching boost instead of moving to a later version with these changes?

File: build-scripts/extras/boost/visualc.hpp.patch (hunk 1 above)
Comment (LamarMoore) [reply build-scripts/extras/boost/visualc.hpp.patch]: @martyngigg It is starting to look like that.

**Comment by martyngigg** [comment]
@quantumsteve We initially had a problem building Boost 1.66 but that's resolved so has been added in #32, which now supersedes this.

**Pull request closed without merging**